earth_enterprise/src/common/CacheSizeCalculations.h

```diff
@@ -19,29 +19,77 @@
 #include "khTypes.h"
 #include <memory>
 #include <vector>
+#include <set>
+#include <map>
+#include <string>
+#include "notify.h"
+#include <qstring.h>
+#include <typeinfo>
+#include <type_traits>
+#include "SharedString.h"
 
-// determine amount of memory used by generic objects
+//determine amount of memory used by an object pointed to by a pointer
 template<class T>
-inline uint64 GetObjectSize(const T &obj) {
-    return sizeof(obj);
+inline uint64 GetHeapUsage(const T* &ptr) {
+    uint64 ptrSize = (ptr ? GetSize(*ptr) : 0);
+    return ptrSize;
 }
-// determine amount of memory used by a vector and all of it's contents
+//determine amount of memory used by an object pointed to by a shared_ptr
 template<class T>
-inline uint64 GetObjectSize(const std::vector<T> &vec) {
+inline uint64 GetHeapUsage(const std::shared_ptr<T> &shptr) {
```

Comment:
Minor: for completeness you might also want a specialization for khRefGuard that's similar to the ptr and shared_ptr specializations. I don't think it matters because we're not using khRefGuards in the asset and asset version caches, though.

Reply:
I had one previously, but removed it since we're not using them, but I can re-implement it.

Reply:
I'm happy either way. On the one hand, it doesn't make sense to include an unnecessary function. On the other, what if we do end up calling GetHeapSize on a khRefGuard at some point in the future and don't notice that it's not implemented? I think I'd come down on the size of leaving it out, but it's up to you. I probably shouldn't have said anything to begin with.